### PR TITLE
Update contract to use diamond storage

### PR DESCRIPTION
**Pull request by Vectorized**
This is a breaking change.

- Delete `scripts/replace-imports.js`.
- Update `scripts/transpile-merge.sh` to use `@gnus.ai/upgrade-safe-transpiler-diamond`.
- Update `package.json` to use `@gnus.ai/upgrade-safe-transpiler-diamond`.
- Removed OpenZeppelin from `package.json`.

Due to the high likelihood of namespace collision when someone uses OpenZeppelin Upgradeable repo with this PR, the following re-namings are automatically done:

- `contracts/Initializable.sol` -> `contracts/ERC721A__Initializable.sol`
- `contracts/InitializableStorage.sol` -> `contracts/ERC721A__InitializableStorage.sol`
- `modifer initializer` -> `modifier initializerERC721A`
- `modifer onlyInitializing` -> `modifier onlyInitializingERC721A`

When inheriting from `ERC721AUpgradeable` and a OpenZeppelin Upgradeable contract, the initializer must have both the `initializerERC721A` and `initializer` modifiers.

```solidity
pragma solidity ^0.8.4;

import 'erc721a-upgradeable/contracts/ERC721AUpgradeable.sol';
import '@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol';

contract Something is ERC721AUpgradeable, OwnableUpgradeable {
  function initialize() initializerERC721A, initializer public {
    __ERC721A_init('Something', 'SMTH');
    __Ownable_init();
  }

  function adminMint(uint256 quantity) external payable onlyOwner {
    // _safeMint's second argument now takes in a quantity, not a tokenId.
    _safeMint(msg.sender, quantity);
  }
}
```

Issue: https://github.com/chiru-labs/ERC721A/issues/267

### CODE REVIEW

**Comment by Vectorized** [comment]
I need to prefix the Initializable classes with `ERC721A__` to prevent name collision in case users import OpenZeppelin or something.

File: contracts/ERC721AStorage.sol [hunk 1]
```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
```
Comment (cygaar) [inline contracts/ERC721AStorage.sol:3]: Do we need to make this 0.8.4 to match ERC721AUpgradeable?

Comment (cygaar) [reply contracts/ERC721AStorage.sol:3]: Same with the other pragmas

File: contracts/ERC721AStorage.sol [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+
+import { ERC721AUpgradeable } from "./ERC721AUpgradeable.sol";
+
+library ERC721AStorage {
+
+  struct Layout {
+
```
Comment (cygaar) [inline contracts/ERC721AStorage.sol:11]: Nit: remove this blank line

Comment (cygaar) [reply contracts/ERC721AStorage.sol:11]: I'll setup a proper linter for 721A and this repo soon

File: contracts/ERC721AStorage.sol [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    // - [128..191] `numberBurned`
+    // - [192..255] `aux`
+    mapping(address => uint256) _packedAddressData;
+
+    // Mapping from token ID to approved address.
+    mapping(uint256 => address) _tokenApprovals;
+
+    // Mapping from owner to operator approvals
+    mapping(address => mapping(address => bool)) _operatorApprovals;
+
```
Comment (cygaar) [inline contracts/ERC721AStorage.sol:49]: Can remove this blank line too

**Review by cygaar — APPROVED** [review]

File: contracts/ERC721AStorage.sol (hunk 2 above)
Comment (Vectorized) [reply contracts/ERC721AStorage.sol]: I added a `(npm run lint:sol) || true` line to the `transpile-merge.sh`.

File: contracts/ERC721AStorage.sol (hunk 1 above)
Comment (Vectorized) [reply contracts/ERC721AStorage.sol:3]: I think we can just leave it at 0.8.0.

**Comment by xpluscal** [comment]
Can you please update the README / docs. Took me some time to find this :) (Current one fails with `contract is not initializing`

**Comment by mataide** [comment]
Remove the ",":

```
pragma solidity ^0.8.4;

import 'erc721a-upgradeable/contracts/ERC721AUpgradeable.sol';
import '@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol';

contract Something is ERC721AUpgradeable, OwnableUpgradeable {
  function initialize() initializerERC721A initializer public {
    __ERC721A_init('Something', 'SMTH');
    __Ownable_init();
  }

  function adminMint(uint256 quantity) external payable onlyOwner {
    // _safeMint's second argument now takes in a quantity, not a tokenId.
    _safeMint(msg.sender, quantity);
  }
}
```

**Pull request merged**